### PR TITLE
<FormField/> - Fix ellipsis

### PR DESCRIPTION
**Pull request by gilmeir**
#3365 

<Text/> adds the required features:
* ellipsis
* tooltip

### CODE REVIEW

File: src/FormField/FormField.scss [hunk 1]
```diff
@@ -46,6 +46,7 @@ $childrenMargin: 6px;
     display: flex;
     align-items: center;
     flex: 1 0 auto;
+    width: 100%;
```
Comment (nitayneeman) [inline src/FormField/FormField.scss]: I think that `max-width` would be better here

File: src/FormField/FormField.visual.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import FormField from './FormField';
+import Input from '../Input';
+
+storiesOf(`FormField`, module).add('Label ellipsis', () => (
```
Comment (nitayneeman) [inline src/FormField/FormField.visual.js]: I would change the structure of this file according to our convention, like:
```javascript
const tests = [
  {
    describe: 'Something',
    its: [
      {
        it: 'Something',
        props: {
        },
      },
    ],
  },
]

tests.forEach(({ describe, its }) => {
  its.forEach(({ it, props }) => {
    storiesOf(`FormField/${describe}`, module).add(it, () => (
      <FormField {...props} />
    ));
  });
});
```

File: src/FormField/FormField.js [hunk 3]
```diff
@@ -159,6 +165,22 @@ class FormField extends React.Component {
...
+  _renderLabel = props => {
+    const { label, labelSize, id } = this.props;
+
+    return (
+      <Text
+        size={labelSize}
+        htmlFor={id}
+        tagName={'label'}
+        data-hook="formfield-label"
+        {...props}
```
Comment (shlomitc) [inline src/FormField/FormField.js]: why?

File: src/FormField/FormField.scss [hunk 4]
```diff
@@ -69,6 +127,16 @@ $childrenMargin: 6px;
   }
 }
 
+.root.small .label * {
```
Comment (shlomitc) [inline src/FormField/FormField.scss]: we never style with `*`. if we must do it, we need to target the direct child.
I'm not a CSS expert, but maybe it's better to apply the styles to the specific elements using a className

File: src/FormField/FormField.scss [hunk 5]
```diff
@@ -31,25 +53,61 @@ $childrenMargin: 6px;
...
+  align-self: center;
+}
+
+.label {
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  overflow: hidden;
+
+  & * {
```
Comment (shlomitc) [inline src/FormField/FormField.scss]: same comment for `*`

File: src/FormField/FormField.scss [hunk 6]
```diff
@@ -31,25 +53,61 @@ $childrenMargin: 6px;
...
+.labelWrapper {
+  display: flex;
+  align-self: center;
+}
+
+.label {
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  overflow: hidden;
```
Comment (shlomitc) [inline src/FormField/FormField.scss]: why was this needed?

File: src/FormField/FormField.scss [hunk 7]
```diff
@@ -31,25 +53,61 @@ $childrenMargin: 6px;
...
+
+.medium .minLabelHeight {
+  line-height: $mediumSize;
+  min-height: $mediumSize;
+}
+
 .minLabelHeight {
-  min-height: $inlineLabelMinHeight;
-  line-height: $inlineLabelMinHeight;
+  overflow: hidden;
```
Comment (shlomitc) [inline src/FormField/FormField.scss]: why is this needed? it should be solved with `Text` applying ellipsis

File: src/FormField/FormField.js [hunk 8]
```diff
@@ -159,6 +165,22 @@ class FormField extends React.Component {
     (labelPlacement === labelPlacements.left ||
       labelPlacement === labelPlacements.right);
 
+  _renderLabel = props => {
```
Comment (shlomitc) [inline src/FormField/FormField.js]: why all props are needed?

File: src/FormField/FormField.js [hunk 9]
```diff
@@ -173,46 +195,42 @@ class FormField extends React.Component {
...
-            <Label
-              appearance="T1"
-              size={labelSize}
-              children={label}
-              for={id}
-              data-hook="formfield-label"
-            />
-
+            {this._renderLabel({
+              ellipsis: true,
```
Comment (shlomitc) [inline src/FormField/FormField.js]: 1. what is the case for not applying ellipsis?
2. better to pronounce it as a business case. for example, if it's an inline label concern, then it should pass this as a flag and internally in the function, apply the ellipsis

File: src/FormField/FormField.visual.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+  );
+};
+
+const test = (it, props) => ({ it, props });
+
+const tests = [
+  {
+    describe: 'Label',
+    its: [
+      test('Ellipsis', {
```
Comment (shlomitc) [inline src/FormField/FormField.visual.js]: test names should affect what you see.
this means - it should show ellipsis in case of long text

File: src/FormField/FormField.visual.js [hunk 11]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import FormFieldComponent from './FormField';
```
Comment (shlomitc) [inline src/FormField/FormField.visual.js]: this should be named FormField

File: src/FormField/FormField.visual.js [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+import Input from '../Input';
+
+const FormField = props => {
```
Comment (shlomitc) [inline src/FormField/FormField.visual.js]: this is a `FormFieldWrapper`

File: src/FormField/FormField.visual.js [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+  return (
+    <div style={{ width: 285, padding: 50 }}>
+      <FormFieldComponent {...props}>
+        <Input />
```
Comment (shlomitc) [inline src/FormField/FormField.visual.js]: this is only one use case. so if the wrapper is general, it should not contain the children.
`<Input/>` is a test configuration, meaning it should come from the props of a give test

**Comment by yaarams** [comment]
@gilmeir please update https://jira.wixpress.com/browse/WCD-8479
when this is ready.

**Comment by shlomitc** [comment]
fixed in this PR https://github.com/wix/wix-style-react/pull/3506
